### PR TITLE
Documentation

### PR DESCRIPTION
**Pull request by mathieuLivebardon**
PR which adds dome doc:
- prePublish command
- jsdoc.js
- NVM part in Developper.md
- Comments of bin scripts

### CODE REVIEW

File: bin/backEnd.js [hunk 1]
```diff
@@ -1,4 +1,12 @@
-/** @file main examples backend */
+/**
+ * @file Sets up an Express backend server for examples, serving static files,
+ * replacing HTML strings, and enabling a game socket service.
+ * The behavior adapts based on the environment mode and uses package.json configuration.
```
Comment (valentinMachado) [inline bin/backEnd.js:4]: maybe you could refer to cross-env to explain the environment mode like => see [cross-env](https://www.npmjs.com/package/cross-env)

File: bin/prePublish.js [hunk 2]
```diff
@@ -45,17 +72,23 @@ const changeVersionPackageJSON = function (path) {
...
 
+// Update version and dependencies for multiple package.json files
 changeVersionPackageJSON('./packages/shared/package.json').then(() => {
   changeVersionPackageJSON('./packages/browser/package.json').then(() => {
     changeVersionPackageJSON('./packages/node/package.json').then(() => {
       changeVersionPackageJSON('./package.json').then(() => {
-        console.log('Start reset');
-        exec('npm run reset')
+        const commandReset = `npm run reset`;
+        console.log('RUN', commandReset);
```
Comment (valentinMachado) [inline bin/prePublish.js:81]: instead of "RUN" "exec" seems more appropiate

File: bin/prePublish.js [hunk 3]
```diff
@@ -45,17 +72,23 @@ const changeVersionPackageJSON = function (path) {
...
+
+        // Execute reset command
+        exec(commandReset)
           .then(printExec)
           .then(() => {
-            exec(
-              `git describe --tags --match v* --abbrev=0 | xargs -I tag sh -c 'git log tag..HEAD --pretty=format:%s > ./docs/static/ChangelogDiff.txt'`
-            ).then(() => {
+            const commandGenerateChangelog = `git describe --tags --match v* --abbrev=0 | xargs -I tag sh -c 'git log tag..HEAD --pretty=format:%s > ./docs/static/ChangelogDiff.txt'`;
+            console.log('RUN', commandGenerateChangelog);
```
Comment (valentinMachado) [inline bin/prePublish.js:88]: idem

File: docs/static/Developers.md [hunk 4]
```diff
@@ -10,6 +10,12 @@ Developing UD-Viz applications requires knowledge about :
 - [three.js](https://threejs.org/)
 - [iTowns](http://www.itowns-project.org)
 
+## NVM
+
+For developpers using node version manager (nvm) tool is advised. nvm allows you to quickly install and use different versions of node via the command line.
```
[the remark this answers is not in the document]
Comment (mathieuLivebardon) [reply docs/static/Developers.md:15]: right

**Review by LorenzoMarnat — APPROVED** [review]

File: bin/backEnd.js [hunk 5]
```diff
@@ -1,11 +1,13 @@
 /**
  * @file Sets up an Express backend server for examples, serving static files,
  * replacing HTML strings, and enabling a game socket service.
- * The behavior adapts based on the environment mode and uses package.json configuration.
- * @requires {@link https://www.npmjs.com/package/@ud-viz/node|@ud-viz/node}
- * @requires {@link https://www.npmjs.com/package/@ud-viz/shared|@ud-viz/shared}
- * @requires {@link https://www.npmjs.com/package/reload|reload}
- * @requires {@link https://www.npmjs.com/package/string-replace-middleware|string-replace-middleware}
+ * The behavior adapts based on the environment mode (NODE_ENV) inject in procosss.env with cross-env package.
```
Comment (valentinMachado) [inline bin/backEnd.js:4]: typo procosss

Comment (mathieuLivebardon) [reply bin/backEnd.js:4]: I went too fast to merge the branch I had not seen the comment. I fixed it in this new PR. 😶

**Pull request merged**